src/main/java/com/flightstats/hub/webhook/WebhookStateReaper.java

```diff
@@ -28,11 +29,13 @@
     }
 
     void delete(String webhook) {
-        log.info("deleting " + webhook);
-        webhookInProcess.delete(webhook);
-        lastContentPath.delete(webhook, WEBHOOK_LAST_COMPLETED);
-        webhookErrorService.delete(webhook);
-        webhookLeaderLocks.deleteWebhookLeader(webhook);
-        log.info("deleted " + webhook);
+        if (HubProperties.isWebHookLeadershipEnabled()) {
```

Comment:
I might opt for short-circuiting this so the if block is negated and the body is just `return`...it reduces nesting a little bit.

Reply:
Yep, agreed.  Done.